Documentation/pre-build-detection

```diff
@@ -0,0 +1,142 @@
+# Pre-build detection guide
```

Comment:
Can you add a `.md` extension to this file, so it is formatted properly in GitHub?

Reply:
That's on me, somehow skipped my mind when creating it. Thanks for pointing it out